### ISSUE TITLE
Toleration boolean values are not correctly quoted by templates

### ISSUE THREAD

**Issue by mcereda** [istio/istio#16661]
**Bug description**

The `_affinity.tp` template files in the helm charts do not quote correctly boolean values even if those have been quoted in the `values.yaml` file.
For example, should I set a `Toleration` value of _true_ for an element in `global.defaultTolerations` in a `values.yaml` file, the template would not quote the value in the resulting yaml file to be applied in kubernetes, leading to Helm not able to deploy the resulting configuration.

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[X] Networking
[X] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

Boolean values should be quoted correctly

**Steps to reproduce the bug**

In file `values.yaml`, in the helm chart, set the following:

```
global:
  defaultTolerations:
  - key: AnyKey
    operator: Equal
    value: "true"
```

then install Istio using Helm and this values file

**Version (include the output of `istioctl version --remote` and `kubectl version`)**

Istio 1.2.4

```
$ istioctl version --remote
client version: version.BuildInfo{Version:"1.1.6", GitRevision:"04850e14d38a69a38c16c800e237b1108056513e-dirty", User:"mcereda", Host:"ip-10-102-1-2.eu-west-1.compute.internal", GolangVersion:"go1.12.7", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.1.6"}
citadel version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
egressgateway version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.2.3"}
egressgateway version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.2.3"}
galley version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
galley version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
ingressgateway version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.2.3"}
ingressgateway version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.2.3"}
pilot version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
pilot version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
policy version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
policy version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
sidecar-injector version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
sidecar-injector version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
telemetry version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
telemetry version: version.BuildInfo{Version:"1.2.4", GitRevision:"c562694ea6e554c2b60d12c9876d2641cfdd917d-dirty", User:"root", Host:"ccdab576-c621-11e9-abca-26bcb80ec4e0", GolangVersion:"go1.12.9", DockerHub:"docker.io/istio", BuildStatus:"Modified", GitTag:"1.2.3"}
```

Helm 2.13.1

```
elm version
Client: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.13.1", GitCommit:"618447cbf203d147601b4b9bd7f8c37a5d39fbb4", GitTreeState:"clean"}
```

Kubectl 1.15.3

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T12:36:28Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.8-eks-a977ba", GitCommit:"a977bab148535ec195f12edc8720913c7b943f9c", GitTreeState:"clean", BuildDate:"2019-07-29T20:47:04Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**

Downloaded from Github, using helm on an EKS cluster in AWS

**Environment where bug was observed (cloud vendor, OS, etc)**

EKS with kubernetes 1.13 on AWS

**Cluster state archive**

Could not give all those information

**Comment by morvencao** [comment]
I think you meat the `defaultNodeSelector` instead of `defaultTolerations` for the following reasons:
1. `defaultTolerations` isn't referred in `_affinity.tpl`, but `defaultNodeSelector` 
2. I have tried your case with istio 1.2.4, it is working well because `defaultTolerations` are referred by deployment directly with `toYaml` function, see: https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml#L220-L225
3. It doesn't work with `defaultNodeSelector` of the following content:
```
global:
  defaultNodeSelector:
    master: "true"
```

Anyway, thank you for reporting this, I have raised a PR: https://github.com/istio/istio/pull/16689 to fix this.
